### PR TITLE
Support for distance predictions

### PR DESCRIPTION
**Pull request by FilomenoSanchez**
@fsimkovic  this is a big update so I understand if you don't have the time/will to review this. Please let me know if you prefer someone from the group to take care of this (possibly @hlasimpk). A summary of what has changed:

1. Distance prediction file support has been added. The following formats are suported: AlphaFold2, Rosetta NPZ, MapPred and CASP RR Mode2
2.  To make this possible three new data structures have been added to `conkit.core` -> `conkit.core.distance.Distance`, `conkit.core.distogram.Distogram` and `conkit.core.distancefile.DistanceFile`. This follows the same architecture used previously used to parse contact maps, and these classes inherit methods and attributes from `conkit.core.contact.Contact`, `conkit.core.ContactMap.ContactMap` and `conkit.core.ContactFile.ContactFile` respectively. This also makes possible to use methods previously created for contact maps  even if a distance file is provided.
	3. `conkit.io.GenericStructureParser` has been updated to create a `conkit.core.distancefile.DistanceFile` instead of a `conkit.core.ContactFile.ContactFile` instance.
	4. I had to make changes to `conkit/io/_io_tools.py` in order to accommodate AlphaFold2 & Rosetta NPZ formats, which are binary.
	5. Copyright date update
	
From this, I think no. 3 and 4 are the most critical as I'm not sure if doing this will break things somewhere else and I would really appreciate your opinion. Also, this is the first of a series of updates related to this. There are another two updates that I have in mind at the moment:

1. Add a conkit.plot method to plot distograms in the form of heatmaps
2. Update documentation to make clear it is now possible to work with distance predictions

These two PRs will be created when the code is ready, I just wanted to keep things tidy by creating a first PR for the core data structures. As I mentioned at the beginning just let me know if you don't want to review this. Also obviously even if you accept to review this you can still decline to review the next two updates.

### CODE REVIEW

**Comment by fsimkovic** [comment]
@FilomenoSanchez I do not have the time to review these changes. I also lack context and thus cannot be a reasonable assessor of the quality of the changes. Lastly, I'm not longer the owner/maintainer of this project, which means someone within the group needs to take over this responsibility.

**Comment by FilomenoSanchez** [comment]
@fsimkovic I understand, if that's the case I'll add @hlasimpk as a reviewer.

File: conkit/io/caspmode2.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        :obj:`~conkit.core.distancefile.DistanceFile`
+
+        """
+
+        hierarchy = DistanceFile(f_id)
+        hierarchy.original_file_format = "CASPRR_MODE_2"
+        _map = Distogram("distogram_1")
+        hierarchy.add(_map)
+
+        for idx, line in enumerate(f_handle.readlines()):
```
Comment (hlasimpk) [inline conkit/io/caspmode2.py:67]: idx doesn't seem to be used here

File: conkit/io/mappred.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        :obj:`~conkit.core.distancefile.DistanceFile`
+
+        """
+
+        hierarchy = DistanceFile(f_id)
+        hierarchy.original_file_format = "MAPPRED"
+        _map = Distogram("distogram_1")
+        hierarchy.add(_map)
+
+        for idx, line in enumerate(f_handle.readlines()):
```
Comment (hlasimpk) [inline conkit/io/mappred.py:70]: idx also doesn't appear to be used here

File: conkit/io/rosetta_npz.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                 (7, 7.5), (7.5, 8), (8, 8.5), (8.5, 9), (9, 9.5), (9.5, 10), (10, 10.5), (10.5, 11), (11, 11.5),
+                 (11.5, 12), (12, 12.5), (12.5, 13), (13, 13.5), (13.5, 14), (14, 14.5), (14.5, 15), (15, 15.5),
+                 (15.5, 16), (16, 16.5), (16.5, 17), (17, 17.5), (17.5, 18), (18, 18.5), (18.5, 19), (19, 19.5),
+                 (19.5, 20), (20, np.inf))
+
+
+class RosettaNpzParser(BinaryDistanceFileParser):
+    """Parser class for rosetta NPZ distance prediction file"""
+
+    def read(self, f_handle, f_id="casp2"):
```
Comment (hlasimpk) [inline conkit/io/rosetta_npz.py:50]: casp2 ?

**Review by hlasimpk — CHANGES_REQUESTED** [review]
Really tidy code over all, I've just noted a few minor things

**Review by hlasimpk — APPROVED** [review]

**Pull request merged**